drivers/serial/uart_nrfx_uarte.c

```diff
@@ -112,6 +112,19 @@ LOG_MODULE_REGISTER(uart_nrfx_uarte, CONFIG_UART_LOG_LEVEL);
 #define UARTE_ANY_LOW_POWER 1
 #endif
 
+BUILD_ASSERT(NRF_GPD_FAST_ACTIVE1 == 0);
+/* Macro must resolve to literal 0 or 1 */
+#define INSTANCE_IS_FAST(unused, prefix, idx, _)					\
+	COND_CODE_1(DT_NODE_HAS_STATUS_OKAY(UARTE(idx)),				\
+		(COND_CODE_1(UTIL_AND(IS_ENABLED(CONFIG_SOC_NRF54H20_GPD),		\
+			     DT_NODE_HAS_PROP(UARTE(idx), power_domains)),		\
+			    (COND_CODE_0(DT_PHA(UARTE(idx), power_domains, id), (1), (0))),\
+			    (0))), (0))
+
+#if UARTE_FOR_EACH_INSTANCE(INSTANCE_IS_FAST, (||), (0))
```

Comment:
```suggestion
#if (UARTE_FOR_EACH_INSTANCE(INSTANCE_IS_FAST, (||), (0))) && defined(CONFIG_SOC_NRF54H20_GPD)
```
Then logic in `INSTANCE_IN_FAST` could be simplified.
Not a blocker though.